demo/site/src/util/graphQLClient.ts

```diff
@@ -34,5 +38,14 @@ export default function createGraphQLClient(previewData?: PreviewData): GraphQLC
 
     return new GraphQLClient(`${process.env.API_URL_INTERNAL}/graphql`, {
         headers,
+        ...gqlOptions,
     });
 }
+
+export const fetch =
+    typeof window === "undefined"
+        ? inMemorySwrCachingFetch //server side: caching fetch
+        : window.fetch; //client side: default fetch
+export const graphqlClient = createGraphQLClient({
```

Comment:
we need a single, shared instance, use this instance instead of creating every time a new one

DI pattern is not possible into getServerSideProps